plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -2102,6 +2104,122 @@ private void testBucketedTable(Session session, HiveStorageFormat storageFormat,
+                        "1274942432",
+                        200,
+                        1),
+                new BucketedFilterTestSetup(
+                        "bigint",
+                        IntStream.range(0, 1000).mapToLong(i -> i + 312739231274942432L).mapToObj(String::valueOf).collect(toImmutableList()),
+                        "312739231274942432",
+                        200,
+                        1),
+                new BucketedFilterTestSetup(
+                        "real",
+                        IntStream.range(0, 1000).mapToDouble(i -> i + 567.123).mapToObj(val -> "REAL '" + val + "'").collect(toImmutableList()),
+                        "567.123",
+                        201,
+                        1),
+                new BucketedFilterTestSetup(
+                        "double",
+                        IntStream.range(0, 1000).mapToDouble(i -> i + 1234567890123.123).mapToObj(val -> "DOUBLE '" + val + "'").collect(toImmutableList()),
+                        "1234567890123.123",
+                        201,
+                        1),
+                new BucketedFilterTestSetup(
+                        "varchar",
+                        IntStream.range(0, 1000).mapToObj(i -> "'test value " + i + "'").collect(toImmutableList()),
+                        "'test value 5'",
+                        200,
+                        1),
+                new BucketedFilterTestSetup(
+                        "varchar(20)",
+                        IntStream.range(0, 1000).mapToObj(i -> "'test value " + i + "'").collect(toImmutableList()),
+                        "'test value 5'",
+                        200,
+                        1),
+                new BucketedFilterTestSetup(
+                        "date",
+                        IntStream.range(0, 1000).mapToObj(i -> "DATE '2020-02-12' + interval '" + i + "' day").collect(toImmutableList()),
+                        "DATE '2020-02-15'",
+                        200,
+                        1),
+                new BucketedFilterTestSetup(
+                        "array<int>",
+                        IntStream.range(0, 1000)
+                                .mapToObj(i -> format("ARRAY[%s, %s, %s, %s]", i + 22767, i + 22768, i + 22769, i + 22770))
+                                .collect(toImmutableList()),
+                        "ARRAY[22767, 22768, 22769, 22770]",
+                        200,
+                        1),
+                new BucketedFilterTestSetup(
+                        "map<double, int>",
+                        IntStream.range(0, 1000)
+                                .mapToObj(i -> format("MAP(ARRAY[%s, %s], ARRAY[%s, %s])", i + 567.123, i + 568.456, i + 22769, i + 22770))
+                                .collect(toImmutableList()),
+                        "MAP(ARRAY[567.123, 568.456], ARRAY[22769, 22770])",
+                        149,
+                        1));
+        return testSetups.stream()
```

Comment:
Let's also include tests for unsupported types like ROW, short decimal, long decimal, short timestamp, long timestamp either here or in another test

Reply:
I can do it but that test is not related to this change because we will get an un-supported error even when we try to create a bucketed table.

Reply:
I'll add a separate commit for this